tests/functional/aws-node-sdk/test/multipleBackend/delete.js

```diff
@@ -64,7 +71,18 @@ describe('Multiple backend delete', () => {
                     `Expected success, got error ${JSON.stringify(err)}`);
                 s3.getObject({ Bucket: bucket, Key: fileObject }, err => {
                     assert.strictEqual(err.code, 'NoSuchKey', 'Expected ' +
-                        'error but got sucess');
+                        'error but got success');
+                    done();
+                });
+            });
+        });
+        it('should delete object from AWS', done => {
+            s3.deleteObject({ Bucket: bucket, Key: awsObject }, err => {
```

Comment:
If two objects with the same key are put in the same bucket, even if the location constraints are different, the old one will be deleted and the new one put in the new location